Team6220_PowerPlay/src/main/java/org/firstinspires/ftc/team6220_PowerPlay/BaseTeleOp.java

```diff
@@ -1,36 +1,52 @@
 
 public abstract class BaseTeleOp extends BaseOpMode {
+    double xPower;
+    double yPower;
+    double tPower;
 
-    // stick curve method
-    public double stickCurve(double input) {
-        return (Math.signum(input) * (Math.pow(input, 2)));
-    }
+    public void driveChassisWithController() {
+        xPower = gamepad1.left_stick_x * (1 - gamepad1.left_trigger * 0.5) * Constants.DRIVE_SPEED_MULTIPLIER;
+        yPower = gamepad1.left_stick_y * (1 - gamepad1.left_trigger * 0.5) * Constants.DRIVE_SPEED_MULTIPLIER;
+        tPower = gamepad1.right_stick_x * (1 - gamepad1.left_trigger * 0.5) * Constants.DRIVE_SPEED_MULTIPLIER;
 
-    public void teleOpDrive(float leftXPosition, float leftYPosition, float turnPosition) {
+        // atan2 determines angle between the two sticks
+        // case for driving the robot left and right
+        if (Math.abs(Math.atan2(gamepad1.left_stick_y, gamepad1.left_stick_x)) > Constants.DRIVE_DEADZONE_DEGREES) {
+            driveWithIMU(xPower, 0.0, tPower);
 
-        // saving the gamepad inputs into variables
-        // booleans for if the buttons are pressed (to use grabber)
-        boolean xIsPressed = gamepad2.x;
-        boolean aIsPressed = gamepad2.a;
+        // case for driving the robot forwards and backwards
+        } else if (Math.abs(Math.atan2(gamepad1.left_stick_x, gamepad1.left_stick_y)) > Constants.DRIVE_DEADZONE_DEGREES) {
+            driveWithIMU(0.0, yPower, tPower);
 
-        // atan2 determines angle between the two sticks
-        if (Math.abs(Math.atan2(leftYPosition, leftXPosition)) > Constants.DEADZONE_ANGLE) {
-            // case for driving the robot left and right
-            driveWithIMU(leftXPosition, 0, turnPosition);
-        } else if (Math.abs(Math.atan2(leftXPosition, leftYPosition)) > Constants.DEADZONE_ANGLE) {
-            // case for driving the robot up and down
-            driveWithIMU(0, leftYPosition, turnPosition);
+        // case for if the deadzone limits are passed, the robot drives normally
         } else {
-            // case for if the deadzone limits are passed, the robot drives normally
-            driveWithIMU(leftXPosition, leftYPosition, turnPosition);
+            driveWithIMU(xPower, yPower, tPower);
+        }
+    }
+
+    public void driveGrabberWithController() {
+        if (gamepad2.x) {  // press x to close
+            driveGrabber(false);
+        } else if (gamepad2.a) {  // press a to open
+            driveGrabber(true);
         }
+    }
+
+    public void driveSlidesWithController() {
+        motorLVSlides.setPower(-gamepad2.left_stick_y);
+        motorRVSlides.setPower(-gamepad2.left_stick_y);
 
-        // grabber open/close method attached to controller buttons
-        if (xIsPressed) {  // press x to close
-            openGrabber(false);
-        } else if (aIsPressed) {  // press a to open
-            openGrabber(true);
+        if (motorLVSlides.getCurrentPosition() < 0) {
```

Comment:
Please use constants instead of magic numbers

Reply:
Those numbers are defined in constants, we just didn't use them as the slides didn't work until last minute